### PR TITLE
Update misc tools

### PR DESCRIPTION
**Pull request by jacob-delgado**

### CODE REVIEW

**Comment by ericvn** [comment]
Making DNM in case the @istio/release-managers-1-18 might be doing a branch cut soon. If the cut is still a while off, we can  merge earlier.

File: docker/build-tools/Dockerfile [hunk 1]
```diff
@@ -119,7 +118,6 @@ RUN mv /usr/bin/gh ${OUTDIR}/usr/bin
 # Build and install a bunch of Go tools
 RUN go install -ldflags="-s -w" google.golang.org/protobuf/cmd/protoc-gen-go@${GOLANG_PROTOBUF_VERSION}
 RUN go install -ldflags="-s -w" google.golang.org/grpc/cmd/protoc-gen-go-grpc@${GOLANG_GRPC_PROTOBUF_VERSION}
-RUN go install -ldflags="-s -w" github.com/uber/prototool/cmd/prototool@${PROTOTOOL_VERSION}
```
Comment (jacob-delgado) [inline docker/build-tools/Dockerfile:122]: See https://github.com/search?q=org%3Aistio%20prototool&type=code

We don't use these in new repos. The tool has been deprecated for over a year.

File: docker/build-tools/Dockerfile [hunk 2]
```diff
@@ -44,31 +44,30 @@ ENV GOIMPORTS_VERSION=v0.1.0
...
 ENV PROTOLOCK_VERSION=v0.14.0
-ENV PROTOTOOL_VERSION=v1.10.0
 ENV SHELLCHECK_VERSION=v0.9.0
 ENV SU_EXEC_VERSION=0.2
-ENV YQ_VERSION=4.33.2
+ENV YQ_VERSION=4.33.3
 ENV KPT_VERSION=v0.39.3
-ENV BUF_VERSION=v1.13.1
-ENV GCLOUD_VERSION=425.0.0
+ENV BUF_VERSION=v1.17.0
```
Comment (jacob-delgado) [inline docker/build-tools/Dockerfile:69]: @ericvn we need to see if this works on mac or if it is still broken

**Comment by jacob-delgado** [comment]
What do you think of removing

```
GOGO_GRPC_GATEWAY_VERSION
PROTOC_GEN_SWAGGER_VERSION
```

eric? I'm fine with helping doing some of the removals and babysitting it. Having these tools not being used is very confusing. As it is the number of containers and layers we build is already super confusing.

If we later want to create an SBOM for the build container the fewer the tools we have the easier it will be to do so. As we want to approach SLSA we'll need to create SBOMs for our build containers.

File: docker/build-tools/Dockerfile [hunk 3]
```diff
@@ -116,15 +115,30 @@ ADD https://github.com/cli/cli/releases/download/v${GH_VERSION}/gh_${GH_VERSION}
 RUN dpkg -i /tmp/gh_${GH_VERSION}_linux_${TARGETARCH}.deb
 RUN mv /usr/bin/gh ${OUTDIR}/usr/bin
 
+# Install protoc-gen-validate
+RUN set -eux; \
```
Comment (jacob-delgado) [inline docker/build-tools/Dockerfile:119]: go get no longer works as [the](https://github.com/bufbuild/protoc-gen-validate/blob/4e1870c281a44e47ae213a5f8bd704b634b0d693/go.mod#L1) references `github.com/envoyproxy/protoc-gen-validate`

**Comment by jacob-delgado** [comment]
> Making DNM in case the @istio/release-managers-1-18 might be doing a branch cut soon. If the cut is still a while off, we can merge earlier.

I created https://github.com/istio/tools/pull/2442 to update the tools for this branch. The only thing we really need updated in that branch is kubectl and the code generation tools.

Let's use this to refine the tools going forward. The `release-1.18` branches have been created. Given that they are running behind schedule we should focus this for `release-1.19`. Cleaning up any tools we no longer need would be a big benefit to the project.

File: docker/build-tools/Dockerfile (hunk 2 above)
Comment (ericvn) [reply docker/build-tools/Dockerfile:69]: I'll build and run `make gen` again to verify Mac behavior. It looks like from my earlier comments the pipeline also broke so shouldn't have been a Mac only issue.

Comment (ericvn) [reply docker/build-tools/Dockerfile:69]: @jacob-delgado I was able to do a `make gen` with the new image I built (failed on the first `buf generate` before):
```
HEAD is now at 51f7666f updating header presence explanation with example (#2762)
buf generate --config tools/proto/buf.yaml --path operator/pkg/ --output operator  --template tools/proto/buf.golang.yaml
buf generate --config tools/proto/buf.yaml --path pkg/dns/ --output pkg  --template tools/proto/buf.golang.yaml
buf generate --config tools/proto/buf.yaml --path pkg/test/echo --output pkg  --template tools/proto/buf.golang.yaml
buf generate --config tools/proto/buf.yaml --path pkg/workloadapi --output pkg  --template tools/proto/buf.golang.yaml
rm manifests/charts/istiod-remote/templates/*
```

**Review by ericvn — APPROVED** [review]

**Pull request merged**